fh-messaging/lib/fhsrv.js

```diff
@@ -177,77 +254,13 @@ var msg = (function() {
 
 
   router.post('/:topic', function(req, res) {
-    connections.num += 1;
-    connections.last = Date.now();
-    var timings_start = Date.now();
-    var numMessages = 1;
-    var topic = req.params.topic;
-    var messages = req.body;
-    var url = req.url;
-    if (messages instanceof Array) {
-      numMessages = messages.length;
-    }
-    logger.info("POST: " + url + " - topic: " + topic + ' - ' + numMessages + ' messages');
-
-    // any auth (millicore api or )
-    if (!(helpers.isValidAuth(req) || topicExcludedFromAuth(topic))) {
-      logger.debug("Post message api called without valid apikey");
-      messaging.deadletter(topic, messages, function(err) {
-        if (err) {
-          logger.error("Error saving unauthorised message: " + JSON.stringify(err));
-        }
-        res.writeHead(403, "Forbidden");
-        res.end("Access Forbidden");
-      });
-      return;
-    }
-    //end request
-    res.json({"message":"accepted messages"});
-
-    // first filter out any messages we already have
-    messaging.filterExistingMessages(topic, messages, function(err, existingMessages, newMessages) {
-      var timings_filtered = Date.now();
-      if (err) {
-        logger.error("Error filtering after " + (timings_filtered - timings_start) + " milliseconds", err);
-        return;
-      }
-      if (newMessages.length > 0) {
-        messaging.logMessage(topic, newMessages, function(err, objects) {
-          var i;
-          if (err) {
-            logger.error("Error filtering after " + (timings_filtered - timings_start) + " milliseconds", err);
-            return;
-          }
-
-          var ids = [];
-          for (i in objects) {
-            if (objects.hasOwnProperty(i)) {
-              var msg = objects[i];
-              ids.push(msg.MD5);
-            }
-          }
-          var m = {};
-          m.newMessages = ids;
-          if (existingMessages.length > 0) {
-            m.existingMessages = existingMessages;
-          }
-          var timings_logged = Date.now();
-          logger.info("Added messages to topic: " + req.params.topic + " new: " + m.newMessages.length + " existing: " + existingMessages.length + ", filtered in " + (timings_filtered - timings_start) + " milliseconds" + ", logged in " + (timings_logged - timings_filtered) + " milliseconds");
-          connections.num -= 1;
-        });
-        var timings_funcLogged = Date.now();
-        logger.info("Call to logMessage() took: " + (timings_funcLogged - timings_filtered) + " milliseconds");
+    limiter.removeTokens(1, function(err, remainingRequests) {
+      if (remainingRequests < 1) {
+        return res.status(429).send('Request Quota Exceeded - 1M/day');
       } else {
-        var m = {};
```

Comment:
What is `m` for? I can't see this in the new `handleTopicPost` function, do we still need this?